Gems/EMotionFX/Code/MCore/Source/CompressedQuaternion.inl

```diff
@@ -31,10 +31,10 @@ MCORE_INLINE TCompressedQuaternion<StorageType>::TCompressedQuaternion(float xVa
 // constructor
 template <class StorageType>
 MCORE_INLINE TCompressedQuaternion<StorageType>::TCompressedQuaternion(const AZ::Quaternion& quat)
-    : m_x((StorageType)(static_cast<float>(quat.GetX()) * CONVERT_VALUE))
-    , m_y((StorageType)(static_cast<float>(quat.GetY()) * CONVERT_VALUE))
-    , m_z((StorageType)(static_cast<float>(quat.GetZ()) * CONVERT_VALUE))
-    , m_w((StorageType)(static_cast<float>(quat.GetW()) * CONVERT_VALUE))
+    : m_x((StorageType)(static_cast<float>(quat.GetX()) * (float)CONVERT_VALUE))
+    , m_y((StorageType)(static_cast<float>(quat.GetY()) * (float)CONVERT_VALUE))
+    , m_z((StorageType)(static_cast<float>(quat.GetZ()) * (float)CONVERT_VALUE))
+    , m_w((StorageType)(static_cast<float>(quat.GetW()) * (float)CONVERT_VALUE))
```

Comment:
You can use `static_cast<float>()` already at the definition of `CONVERT_VALUE` and change it to a `static constexpr float` instead of having it being an anonymous enum. While you're add it, please do so for all occurrences of `CONVERT_VALUE` as it's always only used as float as far as I can see.